### ISSUE TITLE
Crashes in release mode

### ISSUE THREAD

**Issue by ysbaddaden** [ysbaddaden/prax.cr#16]
I noticed that the `prax-binary` built in release mode regularly segfaults with a non helpful message like `segfault in fun literal~42`. I haven't been able to reproduce the problem in non release mode, yet.

There doesn't seem to be a user driven process do reproduce the bug, which just pops up sometimes, either while using Prax or not.

**Comment by ysbaddaden** [comment]
I just got a segfaut using the released 0.4.1 debian package:

Segfault:
```
Segfault happened at: 0x419455 <~fun_literal_43+21>: cmpl $0x1,0x8(%rax)
PC(0x00419455) ok
source "$0x1" ok
destination "0x8(%rax)" (0x00000008) not located in a known VMA region (needed writable region)!
```

Backtrace:
```
#0 ... in ~fun_literal_43()
#1 ... in ~fun_literal_3()
#2 ... in co_runner()
#3 ... in ?? () from /lib/x86_64-linux-gnu/libc/so.6
#4 ... in ?? ()
```

Disassembly:
```
cmpl $0x1,0x8(%rax)
jne ... <~fun_literal_43+616>
...
jne ... <~fun_literal_43+200>
...
callq ... <GC_malloc>
...
callq ... <SSLv23_method@plt>
```

**Comment by ysbaddaden** [comment]
The segfault seems to have been fixed. I didn't experience the segfault for weeks using the latest debian package. Closing :)
